pkg/image/apis/image/helper.go

```diff
@@ -414,6 +414,10 @@ func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *
 			return tag, &tagRef, multiple, nil
 		}
 
+		if tagRef.From.Namespace != "" && tagRef.From.Namespace != stream.ObjectMeta.Namespace {
+			return tag, nil, multiple, ErrCrossImageStreamReference
```

Comment:
Could you fix the docstring of the error as well?

Reply:
Nice catch, fixed.